src/categories/categories.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CategoriesRepository } from './categories.repository';
+import { CreateCategoryDto } from './dto/create-category.dto';
+
+@Injectable()
+export class CategoriesService {
+  constructor(
+    @InjectRepository(CategoriesRepository)
+    private categoryRepository: CategoriesRepository,
+  ) {}
+  async createCategory(
+    createCategoryDto: CreateCategoryDto,
+  ): Promise<CreateCategoryDto> {
+    return this.categoryRepository.createCategory(createCategoryDto);
+  }
+  async deleteCategory(id: string) {
+    const result = await this.categoryRepository.delete({ id });
+    if (result.affected === 0)
+      throw new NotFoundException(`No category with ID ${id} is not found`);
+    return result;
```

Comment:
const result = await this.categoryRepository.findOne({ id });

if (!result) { throw A };
return await this.categoryRepository.delete({ id });

Reply:
done